dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala

```diff
@@ -0,0 +1,1056 @@
+                       remoteLoseSig,
+                       remoteWinSig,
+                       localFundingSigs)
+            }
+          }
+          fundingTx <- getFundingTx
+        } yield {
+          SetupDLC(
+            fundingTx,
+            cetWinLocal,
+            cetWinLocalWitness,
+            cetLoseLocal,
+            cetLoseLocalWitness,
+            cetWinRemote.txIdBE,
+            cetWinRemoteWitness,
+            cetLoseRemote.txIdBE,
+            cetLoseRemoteWitness,
+            refundTx
+          )
+        }
+    }
+  }
+
+  def constructClosingTx(
+      privKey: ECPrivateKey,
+      spendingInfo: BitcoinUTXOSpendingInfoFull,
+      isWin: Boolean,
+      spendsToLocal: Boolean): Future[Transaction] = {
+    // If spendsToLocal, use payout as value, otherwise subtract fee
+    val spendingTxF = if (spendsToLocal) {
+      val payoutValue = if (isWin) {
+        winPayout
+      } else {
+        losePayout
+      }
+
+      val txBuilder = BitcoinTxBuilder(
+        destinations = Vector(
+          TransactionOutput(payoutValue,
+                            P2WPKHWitnessSPKV0(privKey.publicKey))),
+        utxos = Vector(spendingInfo),
+        feeRate = feeRate,
+        changeSPK = emptyChangeSPK,
+        network = network
+      )
+
+      txBuilder.flatMap(_.sign)
+    } else {
+      val txBuilder = BitcoinTxBuilder(
+        destinations = Vector(
+          TransactionOutput(spendingInfo.output.value,
+                            P2WPKHWitnessSPKV0(privKey.publicKey))),
+        utxos = Vector(spendingInfo),
+        feeRate = feeRate,
+        changeSPK = emptyChangeSPK,
+        network = network
+      )
+
+      txBuilder.flatMap(subtractFeeAndSign)
+    }
+
+    spendingTxF.foreach(tx => logger.info(s"Closing Tx: ${tx.hex}"))
+
+    spendingTxF
+  }
+
+  /** Constructs and executes on the unilateral spending branch of a DLC
+    *
+    * @return Each transaction published and its spending info
+    */
+  def executeUnilateralDLC(
+      dlcSetup: SetupDLC,
+      oracleSigF: Future[SchnorrDigitalSignature]): Future[DLCOutcome] = {
+    val SetupDLC(fundingTx,
+                 cetWinLocal,
+                 cetWinLocalWitness,
+                 cetLoseLocal,
+                 cetLoseLocalWitness,
+                 _,
+                 _,
+                 _,
+                 _,
+                 _) = dlcSetup
+
+    oracleSigF.flatMap { oracleSig =>
+      val sigForWin = Schnorr.verify(messageWin, oracleSig, oraclePubKey)
+      val sigForLose = Schnorr.verify(messageLose, oracleSig, oraclePubKey)
+
+      // Pick the CET to use and payout by checking which message was signed
+      val (cet, extCetPrivKey, cetScriptWitness) =
+        if (sigForWin) {
+          (cetWinLocal, cetWinPrivKey, cetWinLocalWitness)
+        } else if (sigForLose) {
+          (cetLoseLocal, cetLosePrivKey, cetLoseLocalWitness)
+        } else {
+          throw new IllegalStateException(
```

Comment:
Do we want to do something like broadcast the refund transaction in this case?

Reply:
Not necessarily, under the current client design (which will change) I would prefer a failure followed later, if wanted, by a call to `executeRefundDLC`